adoc/extensions/sycl_khr_command_graph.adoc

```diff
@@ -52,46 +150,129 @@ enum class node_type {
 class node {
 public:
   node() = delete;
+  
+  /* -- common interface members -- */
 
-  node_type get_type() const;
+  node_type get_type() const noexcept;
 
-  std::vector<node> get_predecessors() const;
+  std::vector<node> get_predecessors() const noexcept;
 
-  std::vector<node> get_successors() const;
+  std::vector<node> get_successors() const noexcept;
 
   static node get_node_from_event(event nodeEvent);
 };
 
 }  // sycl::khr
 ----
 
-:crs: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:reference-semantics
+[[sec:khr-graph-node-member-funcs]]
+==== Member Functions
 
-==== Node Member Functions
+.[apidef]#khr::node::get_type#
+[source, role=synopsis,id=api:khr-node-get-type]
+----
+node_type get_type() const noexcept
+----
 
-==== Depends-On Property
+_Returns:_ A value representing the type of command this node represents.
 
-[source,c++]
+'''
+
+.[apidef]#khr::node::get_predecessors#
+[source, role=synopsis,id=api:khr-node-get-predecessors]
 ----
-namespace sycl::khr::property::node {
-class depends_on {
-  public:
-    template<typename... NodeTN>
-    depends_on(NodeTN... nodes);
-};
-}
+std::vector<node> get_predecessors() const noexcept
 ----
 
-==== Depends-On-All-Leaves Property
-[source,c++]
+_Returns:_ A list of the predecessor nodes which this node directly depends on.
+
+'''
+
+.[apidef]#khr::node::get_successors#
+[source, role=synopsis,id=api:khr-node-get-successors]
 ----
-namespace sycl::khr::property::node {
-class depends_on_all_leaves {
-  public:
-    depends_on_all_leaves() = default;
-};
-}
+std::vector<node> get_successors() const noexcept
+----
+
+_Returns:_ A list of the successor nodes which directly depend on this node.
+
+'''
+
+[[sec:khr-graph-node-static-member-funcs]]
+==== Static member functions
+
+.[apidef]#khr::node::get_node_from_event#
+[source, role=synopsis,id=api:khr-node-get-node-from-event]
 ----
+static node get_node_from_event(event nodeEvent)
+----
+
+_Effects:_ Finds the node associated with an event [code]#nodeEvent# created
+from a submission to a queue in the recording state.
+
+_Returns:_ Graph node that was created when the command that returned
+[code]#nodeEvent# was submitted.
+
+_Throws:_ An [code]#exception# with error code [code]#errc::invalid# if
+[code]#nodeEvent# is not associated with a graph node.
+
+'''
+
+[[sec:khr-node-properties]]
+==== Node properties
+
+The properies that can be provided when adding a [code]#node# to a graph are
+described in <<table.properties.graph.node>>.
+
+[[table.properties.graph.node]]
+.Properties supported by the [code]#node# class
+[width="100%",options="header",separator="@",cols="65%,35%"]
+|====
+@ Property @ Description
+a@
+[source]
+----
+khr::property::node::depends_on
+----
+    a@ The [code]#depends_on# property can be used to specify dependencies for nodes added via
```

Comment:
Can we omit this indentation before `a@`?

Reply:
Done! This was imitating the existing spec reference I was using but I think it doesn't actually matter.